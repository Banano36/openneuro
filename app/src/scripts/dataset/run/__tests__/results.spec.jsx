import React from 'react'
import { shallow } from 'enzyme'
import Results from '../results.jsx'

describe('dataset/run/Results', () => {
  it('should render something when all job results are in', () => {
    const exampleRun = {
      results: [
        {
          type: 'file',
          dirPath: 'avg_brain_size.txt',
          name: 'avg_brain_size.txt',
          path:
            'openneuro.outputs/cf02b5bf17df80d056effadccffd65bd/dcedf0c3-4438-466d-9f85-61c75275b355/avg_brain_size.txt',
        },
      ],
      analysis: {
        jobs: ['only_one_job'],
      },
    }
    expect(
      shallow(
        <Results
          run={exampleRun}
          acknowledgements="acknowledge"
          displayFile={jest.fn()}
          toggleFolder={jest.fn()}
        />,
      ),
    ).toMatchSnapshot()
  })
  it('should render nothing if there are less results than there are jobs', () => {
    const lessResultsRun = {
      results: [
        {
          type: 'file',
          dirPath: 'avg_brain_size.txt',
          name: 'avg_brain_size.txt',
          path:
            'openneuro.outputs/cf02b5bf17df80d056effadccffd65bd/dcedf0c3-4438-466d-9f85-61c75275b355/avg_brain_size.txt',
        },
      ],
      analysis: {
        jobs: ['job_one', 'job_two'],
      },
    }
    expect(
      shallow(
        <Results
          run={lessResultsRun}
          acknowledgements="acknowledge"
          displayFile={jest.fn()}
          toggleFolder={jest.fn()}
        />,
      ).getElement(),
    ).toBe(null)
  })
  it('should render nothing when there are no results', () => {
    const noResultsRun = {
      results: [],
    }
    expect(
      shallow(
        <Results
          run={noResultsRun}
          acknowledgements="acknowledge"
          displayFile={jest.fn()}
          toggleFolder={jest.fn()}
        />,
      ).getElement(),
    ).toBe(null)
  })
})
