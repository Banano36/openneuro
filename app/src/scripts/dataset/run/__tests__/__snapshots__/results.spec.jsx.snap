// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dataset/run/Results should render something when all job results are in 1`] = `
<Accordion
  accordion={true}
  className="results"
>
  <Panel
    bsClass="panel"
    bsStyle="default"
    className="fade-in"
    defaultExpanded={false}
    header="results"
  >
    <div
      className="app-acknowledgements"
    >
      <label>
        Acknowledgements
      </label>
      <div
        className="markdown"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<p>acknowledge</p>
",
          }
        }
      />
    </div>
    <hr />
    <DownloadAll
      run={
        Object {
          "analysis": Object {
            "jobs": Array [
              "only_one_job",
            ],
          },
          "results": Array [
            Object {
              "dirPath": "avg_brain_size.txt",
              "name": "avg_brain_size.txt",
              "path": "openneuro.outputs/cf02b5bf17df80d056effadccffd65bd/dcedf0c3-4438-466d-9f85-61c75275b355/avg_brain_size.txt",
              "type": "file",
            },
          ],
        }
      }
    />
    <div
      className="file-structure fade-in panel-group"
    >
      <div
        className="panel panel-default"
      >
        <div
          aria-expanded="false"
          className="panel-collapse"
        >
          <div
            className="panel-body"
          >
            <FileTree
              displayFile={[Function]}
              editable={false}
              getFileDownloadTicket={[Function]}
              toggleFolder={[MockFunction]}
              tree={
                Array [
                  Object {
                    "dirPath": "avg_brain_size.txt",
                    "name": "avg_brain_size.txt",
                    "path": "openneuro.outputs/cf02b5bf17df80d056effadccffd65bd/dcedf0c3-4438-466d-9f85-61c75275b355/avg_brain_size.txt",
                    "type": "file",
                  },
                ]
              }
            />
          </div>
        </div>
      </div>
    </div>
  </Panel>
</Accordion>
`;
